lisa/tests/base.py

```diff
@@ -1272,14 +1277,15 @@ def run_rtapp(cls, target, res_dir, profile=None, ftrace_coll=None, cg_cfg=None)
                                profile, res_dir=res_dir)
         cgroup = cls._target_configure_cgroup(target, cg_cfg)
         as_root = cgroup is not None
+        wload_cm = wload if wipe_run_dir else nullcontext(wload)
 
         # Pre-hit the calibration information, in case this is a lazy value.
         # This avoids polluting the trace and the dmesg output with the
         # calibration tasks. Since we know that rt-app will always need it for
         # anything useful, it's reasonable to do it here.
         target.plat_info['rtapp']['calib']
 
-        with dmesg_coll, ftrace_coll, target.freeze_userspace():
+        with wload_cm, dmesg_coll, ftrace_coll, target.freeze_userspace():
```

Comment:
We need to make sure we don't add an earlier context manager that tries to do something in the run dir, but AFAICT that should always be contained in the `run()` method. Other than this slippery slope, LGTM

Reply:
Yes, the current code has no issue with the cleanup. Future code will fail quite obviously so we should be fine